Build doesn't pass on master

```shell
git clone https://github.com/ElementsProject/lightning lightning2 && cd lightning2 && ./configure && make -j10
```
Causes
```shell
In file included from ./common/bolt11.h:5,
                 from plugins/autoclean.c:7:
./bitcoin/pubkey.h:9:10: fatal error: secp256k1.h: No such file or directory
 #include <secp256k1.h>
```
Extended : 
```shell
ccan/ccan/tal/path/path.c: In function ‘path_readlink’:
ccan/ccan/tal/path/path.c:325:6: warning: ‘len’ may be used uninitialized in this function [-Wmaybe-uninitialized]
   ret[len] = '\0';
      ^
tools/generate-wire.py --header gossipd/gen_gossip_store.h gossip_store_type < gossipd/gossip_store.csv > gossipd/gen_gossip_store.h
tools/generate-wire.py gossipd/gen_gossip_peerd_wire.h gossip_peerd_wire_type < gossipd/gossip_peerd_wire.csv > gossipd/gen_gossip_peerd_wire.c
tools/generate-wire.py gossipd/gen_gossip_store.h gossip_store_type < gossipd/gossip_store.csv > gossipd/gen_gossip_store.c
Submodule 'daemon/jsmn' (https://github.com/zserge/jsmn) registered for path 'external/jsmn'
Submodule 'external/libbacktrace' (https://github.com/ianlancetaylor/libbacktrace.git) registered for path 'external/libbacktrace'
Submodule 'libsodium' (https://github.com/jedisct1/libsodium.git) registered for path 'external/libsodium'
Submodule 'external/libwally-core' (https://github.com/ElementsProject/libwally-core.git) registered for path 'external/libwally-core'
tools/generate-wire.py connectd/gen_connect_gossip_wire.h connect_gossip_wire_type < connectd/connect_gossip_wire.csv > connectd/gen_connect_gossip_wire.c
tools/generate-wire.py --bolt wire/gen_onion_wire.h onion_type < wire/gen_onion_wire_csv > wire/gen_onion_wire.c
tools/generate-wire.py openingd/gen_opening_wire.h opening_wire_type < openingd/opening_wire.csv > openingd/gen_opening_wire.c
tools/generate-wire.py common/gen_peer_status_wire.h peer_status < common/peer_status_wire.csv > common/gen_peer_status_wire.c
Cloning into '/home/darosior/projects/lightning2/external/jsmn'...
tools/generate-wire.py channeld/gen_channel_wire.h channel_wire_type < channeld/channel_wire.csv > channeld/gen_channel_wire.c
tools/generate-wire.py closingd/gen_closing_wire.h closing_wire_type < closingd/closing_wire.csv > closingd/gen_closing_wire.c
tools/generate-wire.py onchaind/gen_onchain_wire.h onchain_wire_type < onchaind/onchain_wire.csv > onchaind/gen_onchain_wire.c
tools/generate-wire.py connectd/gen_connect_wire.h connect_wire_type < connectd/connect_wire.csv > connectd/gen_connect_wire.c
tools/generate-wire.py --bolt --printwire devtools/gen_print_wire.h wire_type < wire/gen_peer_wire_csv > devtools/gen_print_wire.c
tools/generate-wire.py --bolt --printwire --header devtools/gen_print_wire.h wire_type < wire/gen_peer_wire_csv > devtools/gen_print_wire.h
tools/generate-wire.py --bolt --printwire --header devtools/gen_print_onion_wire.h onion_type < wire/gen_onion_wire_csv > devtools/gen_print_onion_wire.h
tools/generate-wire.py --bolt --printwire devtools/gen_print_onion_wire.h onion_type < wire/gen_onion_wire_csv > devtools/gen_print_onion_wire.c
cc -DBINTOPKGLIBEXECDIR='"'../libexec/c-lightning'"' -Wall -Wundef -Wmissing-prototypes -Wmissing-declarations -Wstrict-prototypes -Wold-style-definition -std=gnu11 -g -fstack-protector -Og -I ccan -I external/libwally-core/include/ -I external/libwally-core/src/secp256k1/include/ -I external/jsmn/ -I external/libbacktrace/ -I external/libbacktrace-build -I . -I/usr/local/include    -DSHACHAIN_BITS=48 -DJSMN_PARENT_LINKS  -DCOMPAT_V052=1 -DCOMPAT_V060=1 -DCOMPAT_V061=1 -DCOMPAT_V062=1 -DCOMPAT_V070=1 -DBINTOPKGLIBEXECDIR='"'../libexec/c-lightning'"'  -c -o plugins/autoclean.o plugins/autoclean.c
In file included from ./common/bolt11.h:5,
                 from plugins/autoclean.c:7:
./bitcoin/pubkey.h:9:10: fatal error: secp256k1.h: No such file or directory
 #include <secp256k1.h>
          ^~~~~~~~~~~~~
compilation terminated.
make: *** [<builtin>: plugins/autoclean.o] Error 1
```

When the command fails, the following is happening after the error as kind of a tear-down:

```
....
 #include <secp256k1.h>
          ^~~~~~~~~~~~~
compilation terminated.
make: *** [<builtin>: plugins/autoclean.o] Error 1
make: *** Waiting for unfinished jobs....
Cloning into '/home/will/tmp/lightning2/external/libbacktrace'...
Cloning into '/home/will/tmp/lightning2/external/libsodium'...

Cloning into '/home/will/tmp/lightning2/external/libwally-core'...
Submodule path 'external/jsmn': checked out '18e9fe42cbfe21d65076f5c77ae2be379ad1270f'
Submodule path 'external/libbacktrace': checked out '5a99ff7fed66b8ea8f09c9805c138524a7035ece'
Submodule path 'external/libsodium': checked out '675149b9b8b66ff44152553fb3ebf9858128363d'
Submodule path 'external/libwally-core': checked out '1b7ce30c8ceb9e95820486dbc1015b04f5f8b683'
```

After this the next `make -j10` compiles fine. So we have a minor Makefile issue here, where the cloning of `external` stuff has to be done first.

--- a/plugins/autoclean.c
+++ b/plugins/autoclean.c
@@ -1,15 +1,9 @@
 #include <ccan/array_size/array_size.h>
-#include <ccan/cast/cast.h>
-#include <ccan/intmap/intmap.h>
 #include <ccan/tal/str/str.h>
 #include <ccan/time/time.h>
-#include <common/amount.h>
-#include <common/bolt11.h>
-#include <common/pseudorand.h>
-#include <common/type_to_string.h>
-#include <gossipd/gossip_constants.h>
+#include <common/utils.h>
+#include <inttypes.h>
 #include <plugins/libplugin.h>
-#include <stdio.h>
 
 static u64 cycle_seconds = 0, expired_by = 86400;
 static struct plugin_timer *cleantimer;

